### PR TITLE
do semver check on containerd + set lowest supported containerd ver

### PR DESCRIPTION
**Pull request by eriksywu**

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_cmd.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ done;
 {{GetInitAKSCustomCloudFilepath}} >> /var/log/azure/cluster-provision.log 2>&1;
 {{end}}
 ADMINUSER={{GetParameter "linuxAdminUsername"}}
-CONTAINERD_VERSION={{GetParameter "containerdVersion"}}
+CONTAINERD_VERSION="1.3.4"
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cse_cmd.sh:15]: 1.3.7

Comment (xuto2) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh:15]: do we need the extra "" ? We can actually remove this line

File: parts/linux/cloud-init/artifacts/cse_helpers.sh [hunk 2]
```diff
@@ -302,4 +302,14 @@ systemctlDisableAndStop() {
         systemctl_disable 20 5 25 $1 || echo "$1 could not be disabled"
     fi
 }
+
+# return true if a >= b 
+semverCompare() {
+    VERSION_A=$1
+    VERSION_B=$2
+    [[ "${VERSION_A}" == "${VERSION_B}" ]] && return 0
+    sorted=( $( echo ${VERSION_A} ${VERSION_B} | tr ' ' '\n' | sort -V ) )
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cse_helpers.sh:311]: I think we should trim the versions before the equality comparison?

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 3]
```diff
@@ -122,13 +122,10 @@ installMoby() {
 # CSE+VHD can dicate the containerd version, users don't care as long as it works
 installContainerd() {
     # we want at least 1.3.x - need to safeguard against aks-e setting this to < 1.3.x
-    if [[ ! "${CONTAINERD_VERSION}" =~ 1\.[3-9]\.[0-9].* ]]; then
-        echo "requested ${CONTAINERD_VERSION} is not supported, setting desired version to 1.3.7"
-        CONTAINERD_VERSION="1.3.7"
-    fi
+    # target CONTAINERD_VERSION is hard-coded to 1.3.4 so this is the lowest supported version
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cse_install.sh:125]: s/1.3.4/1.3.7

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 4]
```diff
@@ -122,13 +122,10 @@ installMoby() {
...
     CURRENT_VERSION=$(containerd -version | cut -d " " -f 3 | sed 's|v||' | cut -d "+" -f 1)
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cse_install.sh:126]: comment for why we need the +?

File: parts/linux/cloud-init/artifacts/cse_cmd.sh [hunk 5]
```diff
@@ -12,7 +12,7 @@ done;
...
 MOBY_VERSION={{GetParameter "mobyVersion"}}
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cse_cmd.sh:16]: can you do the same for the moby version ? in next PR?

File: parts/linux/cloud-init/artifacts/cse_helpers.sh (hunk 2 above)
Comment (eriksywu) [reply parts/linux/cloud-init/artifacts/cse_helpers.sh:311]: yea im assuming the version strings passed in are already trimmed. i can add a trim in here as well to be safe

File: parts/linux/cloud-init/artifacts/cse_cmd.sh (hunk 1 above)
Comment (eriksywu) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh:15]: removing

File: parts/linux/cloud-init/artifacts/cse_cmd.sh (hunk 5 above)
Comment (eriksywu) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh:16]: yea ill make a separate pr for that

**Pull request merged**